dvc/api.py

```diff
@@ -0,0 +1,25 @@
+from dvc.utils.compat import open as _open
+from dvc.repo import Repo
+
+
+def get_url(path, repo_dir=None, remote=None):
+    """Returns an url of `path` in default or specified remote"""
+    repo = Repo(repo_dir)
```

Comment:
Why not just make get_url part of the Repo? What is the point of having it separately, if you initialize repo anyway?

Reply:
To not expose `Repo` as public api.

Reply:
But Repo _is_ a public api. Similar to Repo() from gitpython.

Reply:
As far as I can see our public API is command line only. No internals are documented. And anyway this is the way we discussed it with @shcheklein and @dmpetrov - make it separately, so that users won't need to know about `Repo` and everything.

Reply:
just to clarify - does it accept URL as a repo_dir? Will Repo be responsible for running git clone?

There was a discussion indeed - all 4 of us were involved and we agreed to not expose Repo yet. It also feels that this API is not the same as the Repo one. It's like gitpython vs requests or something. One to deal with DVC CLI on a python level, another one is a higher level API for Python. The difference could be a context manager for an object, or a reading file into memory vs to the disk (the way we do in Repo as I would assume).

Reply:
It doesn't yet support url as repo dir. From what @efiop told me I thought it should be done on top of packages.